command/tls/ca/create/tls_ca_create.go

```diff
@@ -32,13 +33,14 @@ type cmd struct {
 func (c *cmd) init() {
 	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
 	// TODO: perhaps add a -years arg to better capture user intent given that leap years are a thing
-	c.flags.IntVar(&c.days, "days", 1825, "Provide number of days the CA is valid for from now on. Defaults to 5 years.")
-	c.flags.BoolVar(&c.constraint, "name-constraint", false, "Add name constraints for the CA. Results in rejecting "+
-		"certificates for other DNS than specified. If turned on localhost and -domain will be added to the allowed "+
-		"DNS. If the UI is going to be served over HTTPS its DNS has to be added with -additional-constraint. It is not "+
-		"possible to add that after the fact! Defaults to false.")
-	c.flags.StringVar(&c.domain, "domain", "consul", "Domain of consul cluster. Only used in combination with -name-constraint. Defaults to consul.")
-	c.flags.StringVar(&c.clusterID, "cluster-id", "", "ClusterID of the consul cluster, requires -domain to be set as well. When used will set URIs with spiffeid.")
+	c.flags.IntVar(&c.days, "days", 1825, "Number of days the CA is valid for. Defaults to 1825 days (approximately 5 years).")
+	c.flags.BoolVar(&c.constraint, "name-constraint", false, "Enables X.509 name constraints for the CA. "+
+		"If used, the CA only signs certificates for localhost and the domains specified by -domain and -additional-name-constraint. "+
+		"If Consul's UI is served over HTTPS in your deployment, add its DNS name with -additional-constraint. Defaults to false.")
+	c.flags.StringVar(&c.domain, "domain", "consul", "The DNS domain of the Consul cluster that agents are configured with. "+
+		"Defaults to consul. Only used when -name-constraint is set. "+
+		"Additional domains can be passed with -additional-name-constraint.")
+	c.flags.StringVar(&c.clusterID, "cluster-id", "", "ID of the Consul cluster. Sets the CA's URI with the SPIFFEID of the cluster and domain.")
```

Comment:
tiniest nit (non-blocking): should be implied, but maybe mention that domain must be set.

Reply:
Do you think "Defaults to consul" isn't clear enough here? It is an optional field so it's kind of tricky to express "there must be a domain and it will be consul by default"

Reply:
Given that the default behavior is described in "domain", it might still be a good idea to reference the requirement here so the chain of dependency is explicitly clear.
But not a blocker if you think the connection is clear as is.

Reply:
Ah you were referencing cluster ID! Yes agreed that it could be made explicitly clear. Will update